### DISCUSSION TITLE
After migrating from Pages to App, partytown didn't work properly anymore

### DISCUSSION THREAD

**Discussion by srbadni** [vercel/next.js#73386] [App Router]
I had this code at the bottom of my page in Pages Router:

```
<ErrorBoundary>
 {
 schemaFaq
 ? <Script id={'question-script'} strategy="beforeInteractive" type="application/ld+json">
 {schemaFaq}
 </script>
 : null
 }
 </ErrorBoundary>

 <Script
 strategy="worker"
 src={`https://www.googletagmanager.com/gtag/js?id=${process.env.NEXT_PUBLIC_GTAG}`}
 />

 <script
 type="text/partytown"
 id={"googletagmanager"}
 dangerouslySetInnerHTML={{
 __html: `
 window.dataLayer = window.dataLayer || [];
 window.gtag = function gtag(){window.dataLayer.push(arguments);}

 gtag('consent', '${isAcceptCookie ? 'update' : 'default'}', {
 'ad_user_data': '${isAcceptCookie ? 'granted' : 'denied'}',
 'ad_personalization': '${isAcceptCookie ? 'granted' : 'denied'}',
 'ad_storage': '${isAcceptCookie ? 'granted' : 'denied'}',
 'analytics_storage': '${isAcceptCookie ? 'granted' : 'denied'}'
 ${isAcceptCookie ? '' : ", 'wait_for_update': 500"}
 });

 gtag('js', new Date());
 gtag('config', '${process.env.NEXT_PUBLIC_GTAG}');
 `,
 }}
 />
```

And with this script, google tag manager and its operations were going in another thread.
But after migrating to App Router, no script is taken on the network, and no separate doc is created by partytown. Is there a part that I did not consider for this migration?

**Comment by icyJoseph** [comment]
AFAIK, the worker strategy on script, is not yet supported in App Router, as documented here:

- https://nextjs.org/docs/app/api-reference/components/script#worker

<img width="685" alt="Screenshot 2024-11-30 at 21 38 27" src="https://github.com/user-attachments/assets/67968460-ee3a-474d-95b6-e624c2c7ade3">

**Reply by srbadni** [reply]
Oh, I saw it. So what is currently the best alternative to moving the googletagmanager implementation outside of the main thread? to reduce TBT time
